docs/usage.md

```diff
@@ -287,3 +288,104 @@ podman build . \
+
+For enabling an alternative crates.io registry the only required env is 
+
+```
+CARGO_REGISTRIES_MY_REGISTRY_INDEX=https://my-intranet:8080/git/index
+```
+
+[cargo docs on using an alternate registry](https://doc.rust-lang.org/cargo/reference/registries.html#using-an-alternate-registry)
+
+Unfortunately I'm still clueless on how to run an alternative registry (and this don't seem to align with cachi2 ethos), so I'm skipping this option.
+
+#### Replacing crates.io with a local path
+
+TLDR: not possible through environment variables; will need to inject a config file to a specific location
+
+Unfortunately this method does not work with environment variables (see [this cargo issue](https://github.com/rust-lang/cargo/issues/5416)) and requires a .cargo/config.toml file.
+
+It must be placed somewhere relative to where "cargo install/build" will run following the hierarchical structure below
+
+>Cargo allows local configuration for a particular package as well as global configuration. It looks for configuration files in the current directory and all parent directories. If, for example, Cargo were invoked in `/projects/foo/bar/baz`, then the following configuration files would be probed for and unified in this order:
+>-   `/projects/foo/bar/baz/.cargo/config.toml`
+>-   `/projects/foo/bar/.cargo/config.toml`
+>-   `/projects/foo/.cargo/config.toml`
+>-   `/projects/.cargo/config.toml`
+>-   `/.cargo/config.toml`
+>-   `$CARGO_HOME/config.toml` which defaults to:
+ >   -   Windows: `%USERPROFILE%\.cargo\config.toml`
+>    -   Unix: `$HOME/.cargo/config.toml`
+
+Source: [cargo docs](https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure)
+
+This is how I configured mine:
+
+cargo_config.toml
+```
+[source.crates-io]
+replace-with = "local"
+
+[source.local]
+local-registry = "/tmp/cachi2-output"
+```
+local-registry also requires a index of all packages downloaded exactly like what's in https://github.com/rust-lang/crates.io-index. For this POC I just git cloned this repo locally and placed it at `./cachi2-output/index` (NOTE: this is definitely overkill; the final version should use only indexes for the packages we have)
+
+```shell
+git clone git@github.com:rust-lang/crates.io-index.git --depth 1 cachi2-output/index
+```
+
+There's a [cargo local registry package](https://crates.io/crates/cargo-local-registry) that prepares  dependencies like this, but it is pretty buggy and fails for some packages I played around (including the one used on this PoC).
+
+### Build the base image (cargo)
+
+
+Containerfile.baseimage
+```Dockerfile
+FROM registry.redhat.io/ubi9/ubi
+
+RUN dnf install cargo rust rust-std-static -y &&\
+    dnf clean all
+```
+
+```shell
+podman build --tag bat-base-image -f Containerfile.baseimage .
+```
+
+### Build the application image (cargo)
+
+Containerfile
+```Dockerfile
+FROM bat-base-image:latest
+
+COPY cargo_config.toml /app/.cargo/config.toml
```

Comment:
For pure rust projects it is fine to add cargo config to project folder, but this approach didn't work when building with PIP and I had to place it on `/.cargo/config.toml` instead.